src/components/courses/Course.js

```diff
@@ -10,19 +10,43 @@ class Lesson extends Component {
     nextLesson = () => {
         const currentIndex = this.props.courses.currentIndex;
         const lessons = this.props.course.lessons;
-        if (window.confirm("Are you sure you want to continue?\nYou will lose any unsaved work!")) {
+        if(this.isEditorChange()){
+            if (window.confirm("Are you sure you want to continue?\nYou will lose any unsaved work!")) {
+                this.props.courseActions.nextLesson(currentIndex, lessons[currentIndex + 1]);
+            }
+        }else{
             this.props.courseActions.nextLesson(currentIndex, lessons[currentIndex + 1]);
         }
     }
 
     lastLesson = () => {
         const currentIndex = this.props.courses.currentIndex;
         const lessons = this.props.course.lessons;
-        if (window.confirm("Are you sure you want to continue?\nYou will lose any unsaved work!")) {
+        if(this.isEditorChange()){
+            if (window.confirm("Are you sure you want to continue?\nYou will lose any unsaved work!")) {
+                this.props.courseActions.previousLesson(currentIndex, lessons[currentIndex - 1]);
+            }
+        }else{
             this.props.courseActions.previousLesson(currentIndex, lessons[currentIndex - 1]);
         }
     }
 
+    isEditorChange = () => {
+        let text;
+        try {
+            let editor = window.ace.edit("ace-editor");
```

Comment:
This part of the code will only work when user transition between the lessons, but it will call window listener if users try to do anything to do with the refreshing or closing the window